src/testNode.ts

```diff
@@ -0,0 +1,31 @@
+export class TestNode {
+    private _isError: boolean;
+
+    constructor(private _parentPath: string, private _name: string, private _children?: TestNode[]) {
+    }
+
+    public get name(): string {
+        return this._name;
+    }
+
+    public get fullName(): string {
+        return `${this._parentPath}.${this._name}`;
```

Comment:
If `parentPath` is empty (when it is namespace Node), the full name should not have a prefix dot.